meson.build

```diff
@@ -1013,58 +1013,72 @@ conf.set10('HAVE_LIBBPF', libbpf.found())
+                        endif
+                        # Assume that the required flags are supported by the found clang.
+                        clang_supports_flags = clang_found
+                else
+                        clang_found = true
+                        clang = cc.cmd_array()
+                        clang_supports_flags = cc.has_argument('-Wno-compare-distinct-pointer-types')
                 endif
-                # Assume that the required flags are supported by the found clang.
-                clang_supports_flags = clang_found
-        else
-                clang_found = true
-                clang = cc.cmd_array()
-                clang_supports_flags = cc.has_argument('-Wno-compare-distinct-pointer-types')
-        endif
 
-        if clang_found
-                # Check if 'clang -target bpf' is supported.
-                clang_supports_bpf = run_command(clang, '-target', 'bpf', '--print-supported-cpus', check : false).returncode() == 0
-        else
-                clang_supports_bpf = false
+                if clang_found
+                        # Check if 'clang -target bpf' is supported.
+                        clang_supports_bpf = run_command(clang, '-target', 'bpf', '--print-supported-cpus', check : false).returncode() == 0
+                endif
+        elif bpf_compiler == 'gcc'
+                warning('GCC BPF Compiler support is experimental and not recommended.')
+                bpf_gcc = find_program('bpf-gcc',
+                                       required : true,
+                                       version : '>= 12.1.0')
+                bpf_gcc_found = bpf_gcc.found()
         endif
 
-        # Debian installs this in /usr/sbin/ which is not in $PATH.
-        # We check for 'bpftool' first, honouring $PATH, and in /usr/sbin/ for Debian.
-        # We use 'bpftool gen' subcommand, it was added by 985ead416df39d6fe8e89580cc1db6aa273e0175 (v5.6).
-        bpftool = find_program('bpftool',
-                               '/usr/sbin/bpftool',
-                               required : false,
-                               version : '>= 5.13.0')
-
-        if bpftool.found()
-                bpftool_strip = true
-        else
-                bpftool_strip = false
+        if clang_supports_bpf or bpf_gcc_found
+                # Debian installs this in /usr/sbin/ which is not in $PATH.
+                # We check for 'bpftool' first, honouring $PATH, and in /usr/sbin/ for Debian.
+                # We use 'bpftool gen' subcommand, it was added by 985ead416df39d6fe8e89580cc1db6aa273e0175 (v5.6).
                 bpftool = find_program('bpftool',
                                        '/usr/sbin/bpftool',
-                                       required : bpf_framework_required,
-                                       version : '>= 5.6.0')
-        endif
+                                       required : false,
+                                       version : '>= 5.13.0')
```

Comment:
Could you update the comment why we need 5.13.0 here?

Reply:
Comments updated/added.